### DISCUSSION TITLE
I GOT OBTAINIUM ON MY GALAXY WATCH 4

### DISCUSSION THREAD

**Discussion by NicholasFlamy** [ImranR98/Obtainium#1470] [Show and tell]
So I used Wear Installer 2 and installed Obtainium. Then I was trying to figure out how to import profiles. So I first had to `adb push /storage/emulated/0/Backups/Obtainium/obtainiumbackuponphone.json /storage/emulated/0/Documents/obtainiumbackupnowonwatch.json`
Then I had to find an app that can act as a file picker since WearOS doesn't have the AOSP File Picker. The apk for the File Picker on apkmirror once installed would crash when opened on the watch. So I found [Anexplorer Watch File Manager](https://play.google.com/store/apps/details?id=dev.dworks.apps.anexplorer.free) on Google Play and it can open and select files when you tap "Obtainium Import" The problem is that the file manager can only see images, videos, audio, and maybe something else but not general files like  txt and json. So I renamed the file on the watch using `mv obtainiumbackupnowonwatch.json obtainiumbackupnowonwatch.jpg` and then tried import and was able to import the file with the jpg extension. And I was very happy to see that it read the file and imported it regardless of file extension. So @ImranR98 if you ever decide to make it only read json and txt file please consider a switch that allows it to read from a file with the wrong extension. Or just don't make it care about file extension. It's great as it is especially for edge cases.

**Comment by NicholasFlamy** [comment]
I should mention that regular app installing won't work but root or Shizuku likely does. I still need to test Shizuku with it on the watch.

**Comment by einarpersson** [comment]
I don't really understand - is it enought to just install Obtainium using Wear installer 2 in order to use it, or are the subsequent steps described here required to get it to work?

**Reply by NicholasFlamy** [reply]
Wear Installer 2 will get you Obtainium. If you root your watch or use Shizuku, haven't tested yet, then you should be able to install stuff. The other steps were because I didn't want to have to add apps on the watch (it's kinda a pain) and rather just import my config from my phone.

**Reply by einarpersson** [reply]
Why can I install stuff through Obtainium on my Samsung 21 _without_ root, but when it comes to a Smart Watch it (root/Shizuku) is required? Grateful for any clarification (I am farily new to the Android ecosystem).

**Reply by NicholasFlamy** [reply]
The ability to sidelpad an apk onto Android which has a permission for an app to do it has been removed from WearOS (WearOS is Android but with things changed, including some permissions and functions entirely removed.)

**Reply by einarpersson** [reply]
thats really a shame. Thanks for explaining!
